dms/models/dms_file.py

```diff
@@ -644,7 +644,12 @@ def _inverse_content(self):
 
     def _create_model_attachment(self, vals):
         res_vals = vals.copy()
-        directory = self.env["dms.directory"].sudo().browse(res_vals["directory_id"])
+        directory = False
+        directory_model = self.env["dms.directory"].sudo()
+        if "directory_id" in res_vals:
+            directory = directory_model.browse(res_vals["directory_id"])
+        elif self.env.context.get("active_id"):
+            directory = directory_model.browse(self.env.context.get("active_id"))
```

Comment:
Perhaps you could save `self.env.context.get("active_id")` in a variable to avoid having to fetch it twice?
```suggestion
        active_id_from_ctx = self.env.context.get("active_id")
        if "directory_id" in res_vals:
            directory = directory_model.browse(res_vals["directory_id"])
        elif active_id_from_ctx:
            directory = directory_model.browse(active_id_from_ctx)
```

Reply:
This is not bad per se, but it's not gonna give you any more performance. 🤷🏼‍♂️

Reply:
Yes, of course. It is just a small suggestion, not really important :smile:

Reply:
Is more readable?
<pre>
directory_id = res_vals.get("directory_id", self._context.get("active_id"))
directory = directory_id and self.env["dms.directory"].browse(directory_id)
</pre>